chainer_chemistry/dataset/parsers/csv_file_parser.py

```diff
@@ -172,3 +182,7 @@ def parse(self, filepath, return_smiles=False):
             if self.postprocess_fn is not None:
                 result = self.postprocess_fn(result)
             return {"dataset": NumpyTupleDataset(result), "smiles": smileses}
+
+    def extract_total_num(self, filepath):
+        df = _read_csv(filepath)
+        return len(df)
```

Comment:
When `parse` method failed feature extraction from some rows in a CSV file, the final dataset size is different from the value this method returns. Whether it is acceptable or not depends on how we use this method, I would like to know your expected use case.

Reply:
I asked @corochann directly and was told that he expect we use this method to determine the value fed to `target_index` option of `CSVFileParser.parse`. For example, if we want to extract input feature from 10% of whole dataset, we need to know how many samples are in a file. Therefore, the returned value of `extract_total_num` need not to be same as the final dataset size.